tests/unit/check_terminal.c

```diff
@@ -0,0 +1,266 @@
+
+    // connect
+    mock_host_controller_expect_diaddr_req(mock_hostmod_diaddr);
+
+    rv = osd_terminal_connect(terminal_ctx);
+    ck_assert_int_eq(rv, OSD_OK);
+
+    // describe
+    mock_host_controller_expect_mod_describe(mock_hostmod_diaddr,
+                                             mock_dem_uart_diaddr,
+                                             OSD_MODULE_VENDOR_OSD,
+                                             OSD_MODULE_TYPE_STD_DEM_UART, 0);
+
+    // set event dest
+    mock_host_controller_expect_reg_write(
+        mock_hostmod_diaddr, mock_dem_uart_diaddr, OSD_REG_BASE_MOD_EVENT_DEST,
+        mock_hostmod_diaddr);
+
+    // activate module
+    mock_host_controller_expect_reg_read(mock_hostmod_diaddr,
+                                         mock_dem_uart_diaddr,
+                                         OSD_REG_BASE_MOD_CS, 0);
+    mock_host_controller_expect_reg_write(mock_hostmod_diaddr,
+                                          mock_dem_uart_diaddr,
+                                          OSD_REG_BASE_MOD_CS, 1);
+
+    rv = osd_terminal_start(terminal_ctx);
+    ck_assert_int_eq(rv, OSD_OK);
+}
+
+void teardown_hostmod()
+{
+    osd_result rv;
+
+    // deactivate event sending
+    mock_host_controller_expect_reg_read(mock_hostmod_diaddr,
+                                         mock_dem_uart_diaddr,
+                                         OSD_REG_BASE_MOD_CS, 1);
+    mock_host_controller_expect_reg_write(mock_hostmod_diaddr,
+                                          mock_dem_uart_diaddr,
+                                          OSD_REG_BASE_MOD_CS, 0);
+
+    rv = osd_terminal_stop(terminal_ctx);
+    ck_assert_int_eq(rv, OSD_OK);
+
+    rv = osd_terminal_disconnect(terminal_ctx);
+    ck_assert_int_eq(rv, OSD_OK);
+
+    osd_terminal_free(&terminal_ctx);
+    ck_assert_ptr_eq(terminal_ctx, NULL);
+
+    osd_log_free(&log_ctx);
+}
+
+/**
+ * Test fixture: setup (called before each tests)
+ */
+void setup(void)
+{
+    mock_hostmod_diaddr = osd_diaddr_build(1, 1);
+    mock_dem_uart_diaddr = osd_diaddr_build(target_subnet_addr, 15);
+
+    mock_host_controller_setup();
+    setup_hostmod();
+
+    fprintf(stderr, "Setup completed!\n");
+    fprintf(stderr, "hostmod: 0x%x, dem-uart: 0x%x\n", mock_hostmod_diaddr, mock_dem_uart_diaddr);
+}
+
+/**
+ * Test fixture: setup (called after each test)
+ */
+void teardown(void)
+{
+    fprintf(stderr, "Beginning Teardown!\n");
+
+    mock_host_controller_wait_for_event_tx();
+    teardown_hostmod();
+    mock_host_controller_teardown();
+}
+
+START_TEST(test_basic_init)
+{
+    setup();
+    teardown();
+}
+END_TEST
+
+// This test fails for the following chars
+// Sent - Read
+// (0x00..0x1d) - 0x1d (ASCII Group Separator?)
+// 0x7e - 0x80 (~ and non-ascii)
+// 0x7f - 0x80 (DEL & non-ascii)
+
+// If canonical mode and signals are disabled (in terminal.c)
+// 0x0a00 - 0x0d00 (CR & LF switched)
+// 0x0d00 - 0x0a00
+// 0x1100 - 0x1200
```

Comment:
0x11 is XOFF (software flow control), it could be that some termios option related to flow control somehow influences this behavior. I wouldn't worry to much about it, however. It's fine to simply exclude this character (and possibly other rare control signals) from the test.